### DISCUSSION TITLE
Can not add NGXFlickingModule to standalone component angular 14

### DISCUSSION THREAD

**Discussion by hshahi** [naver/egjs-flicking#759] [General]
Hi

I Can not add NGXFlickingModule to standalone component angular 14 import as it complains that 
component import must standalone component, pipes, directives or NGModules

Any advise appreciated
Thanks
Shahi

**Comment by malangfox** [comment]
Hello @hshahi.

We found that importing NgxFlickingModule with standalone component in the latest version of Angular doesn't work properly.
We will fix this bug and will leave a comment when a patch is released.

Thank you for reporting.

**Reply by hshahi** [reply]
Hi, I am just wondering if there is a timeline for releasing this upgrade?
Thanks

**Comment by malangfox** [comment]
In the process of figuring out the environment where this issue occurred, we have noticed updating some packages related to the latest version of Angular solves this issue.
When using the following package versions with the latest version of Angular, the issue with the standalone component seems not to occur.
```
  "dependencies": {
    "@angular/animations": "^15.0.2",
    "@angular/common": "^15.0.2",
    "@angular/compiler": "^15.0.2",
    "@angular/core": "^15.0.2",
    "@angular/forms": "^15.0.2",
    "@angular/platform-browser": "^15.0.2",
    "@angular/platform-browser-dynamic": "^15.0.2",
    "@angular/router": "^15.0.2",
    "@egjs/ngx-flicking": "^4.10.4",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^15.0.2",
    "@angular/cli": "~15.0.2",
    "@angular/compiler-cli": "^15.0.2",
    "@types/jasmine": "~4.0.0",
    "jasmine-core": "~4.3.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.0.0",
    "typescript": "~4.8.4"
  }
```
We are looking for a fine solution to this issue, and would like to know the versions of the packages you are using.
Can we know which package versions you are using?

**Comment by hshahi** [comment]
Hi
Thanks, it is version 14.2.2, But I am thinking to move up to 15

**Comment by hshahi** [comment]
May I kindly ask if you can send me a code sandbox example that uses Angular 15 and at the same time would have arrows.
once thing that would I be able to use and it is a requirement is that the carousel should be able to resize when window resize and hence responsive and when 
carousel resize the number of items with in that is showing and  that each are fix size will change due to the fact that viewport of the carousel has changed. all the examples are fixed size. so when for example we 10 item and when page loads is in max width  for example it only shows 6 and on the right hand side 4 is hidden. when we make the browser or container smaller it should starts showing 5 then 4 and if it is showing item 4, 5, 6, 7 from 1-10, when it resizes it s view port and shows only for ex 4,5,6. since I have not been able
achieve that yet can you come up with advise me how to achieve this. Basically responsive and not only show 1 item but more item in its view.

**Comment by malangfox** [comment]
I'd like to make you a demo of what you want, but it looks like this will need some time.
Before making something, I would like to ask you detailed questions about the implementation you want.

If the number of visible items is different depending on the size of the viewport, it is like an implementation using the `panelsPerView` option.
We have one [demo](https://codepen.io/malangfox/pen/ZEoWgoq) where the option changes depending on the window size.
However, this demo is not implemented on Angular.

Can we check what you want is a demo where the `panelsPerView` option changes according to the horizontal size of the viewport, and an Arrow plugin is added to it?
If we know exactly what you want, we'll be happy to make a demo for you, but this will need some time.

**Comment by hshahi** [comment]
Hi, Thanks

Basically when the size of browser changes and so the carousel itself so if it is maximised horizontally and carousel shows 6 out 10, wheter showing (1,2,3,4,5,6) or has already move forward showing (3,4,5,67,8) then as you reduce the width of browser then 6 changes showing 5 and then 4  and so on. basically fully responsive. The items are fixed width so that changing width of carousel  wont change the size of items but only number of items that are showing inside visible part depending on where you are showing 6 or 5 items from the beginning (1,2,3,4,5) or you have moved forward showing (3,4,5,6,7) or (3,4,5) and so on. Also with arrow. thank you for listening.
